linker/linker_test.go

```diff
@@ -2956,6 +2956,27 @@ func TestLinkerValidation(t *testing.T) {
 			},
 			expectedErr: `test.proto:14:25: expected extension with number 1 to be named .foo.bar, not foo.s, per declaration at test.proto:8:25`,
 		},
+		"failure_extension_name_does_not_match_declaration2": {
+			input: map[string]string{
+				"test.proto": `
+					syntax = "proto2";
+					package foo;
+					message A {
+						extensions 1 [
+							declaration={
+								number: 1
+								full_name: ".foo.bar"
+								type: "string"
+							}
+						];
+					}
+					extend A {
+						optional string s = 1;
+					}
+				`,
+			},
+			expectedErr: `test.proto:13:25: expected extension with number 1 to be named .foo.bar, not foo.s, per declaration at test.proto:7:25`,
```

Comment:
Should this be not `.foo.s`?

Reply:
I don't think the leading dot is helpful. In fact, maybe the message should strip it from the expected type name.

Reply:
Pushed another commit so that all names referenced in these messages are uniform (and do not include leading dot).

Reply:
The fullname for the declaration must include the leading dot. So does this not make it more obscure? What is the error message if the name isn't prefixed correctly?

Reply:
https://github.com/bufbuild/protocompile/pull/303/files#diff-be56fc6b2428e69e2ae94765983ec0f1253a3c5c3fbd2f56a20bbf24910421f8R2688

Reply:
I definitely don't think there is anything obscure about omitting the leading dot. _None_ of the other error messages from the compiler that refer to a type use a leading dot, and I'm pretty sure it's not confusing to users.

The declarations themselves require a dot mainly for disambiguation reasons: they are not allowed to be relative/partially-qualified references, and the leading dot guarantees that.

> What is the error message if the name isn't prefixed correctly?

https://github.com/bufbuild/protocompile/blob/main/linker/linker_test.go#L2688
```
test.proto:7:25: extension declaration full name "foo.bar" should start with a leading dot (.)
```

https://github.com/bufbuild/protocompile/blob/main/linker/linker_test.go#L2741
```
test.proto:8:25: extension declaration type "foo.Bar" must be a builtin type or start with a leading dot (.)
```

Reply:
Resolved the comment, this approach looks good.